src/storage.ts

```diff
@@ -232,51 +223,57 @@ class GoogleCloudStorageHandler implements IPackageStorageManager {
       const packageJson: Package = JSON.parse(content[0].toString('utf8'));
-      this.logger.info(`gcloud: [_readPackage] providing '${PACKAGE_JSON}' file for package '${packageName}' from CACHE`);
+      this.logger.info(
+        `gcloud: [_readPackage] providing '${PACKAGE_JSON}' file for package '${packageName}' from CACHE`
+      );
       return packageJson;
     }
 
     const file = this._buildFilePath(packageName, PACKAGE_JSON);
     try {
       const fileDownloadPromise = file.download();
-      GoogleCloudStorageHandler.packageJsonCache[packageName] = { timestamp: new Date().getTime(), content: fileDownloadPromise };
+      GoogleCloudStorageHandler.packageJsonCache[packageName] = {
+        timestamp: new Date().getTime(),
+        content: fileDownloadPromise,
+      };
       const content: DownloadResponse = await fileDownloadPromise;
-      this.logger.info(`gcloud: [_readPackage] successfully retrieved '${PACKAGE_JSON}' file for package '${packageName}' from storage`);
+      this.logger.info(
+        `gcloud: [_readPackage] successfully retrieved '${PACKAGE_JSON}' file for package '${packageName}' from storage`
+      );
       const packageJson: Package = JSON.parse(content[0].toString('utf8'));
-      this.logger.trace(`gcloud: [_readPackage] '${PACKAGE_JSON}' file for package '${packageName}': ${JSON.stringify(packageJson)}`)
+      this.logger.trace(
+        `gcloud: [_readPackage] '${PACKAGE_JSON}' file for package '${packageName}': ${JSON.stringify(packageJson)}`
+      );
       GoogleCloudStorageHandler.packageJsonCache[packageName] = { timestamp: new Date().getTime(), content: content };
-      return packageJson
+      return packageJson;
     } catch (error) {
-      this.logger.error(`gcloud: [_readPackage] failed retrieving '${PACKAGE_JSON}' file for package '${packageName}' from storage: ${error.message}`);
+      this.logger.error(
+        `gcloud: [_readPackage] failed retrieving '${PACKAGE_JSON}' file for package '${packageName}' from storage: ${error.message}`
+      );
       throw error;
     }
   }
 
-  private _fileExist(name: string, fileName: string): Promise<boolean> {
-    return new Promise(
-      async (resolve, reject): Promise<void> => {
-        const file: File = this._buildFilePath(name, fileName);
-        try {
-          const data = await file.exists();
-          const exist = data[0];
-
-          resolve(exist);
-          this.logger.debug({ name: name, exist }, 'gcloud: check whether @{name} exist successfully: @{exist}');
-        } catch (err) {
-          this.logger.error(
-            { name: file.name, err: err.message },
-            'gcloud: check exist package @{name} has failed, cause: @{err}'
-          );
+  private async _fileExist(name: string, fileName: string): Promise<boolean> {
+    const file: File = this._buildFilePath(name, fileName);
+    try {
+      const data = await file.exists();
+      const exist = data[0];
```

Comment:
NIT
```suggestion
      const [exist] = data;
```

Reply:
Not directly related to this PR, but changed to:

```diff
-      const data = await file.exists();
-      const exist = data[0];
+      const [exist] = await file.exists();
```